Customizing/global/plugins/Services/Repository/RepositoryObject/ReportEduBio/classes/class.ilObjReportEduBioGUI.php

```diff
@@ -79,81 +79,62 @@ public function renderReport() {
+				.($this->spacer !== null ? $this->spacer->render() : "")
+				.($this->object->getWBD()->getWBDTPType() !== gevWBD::WBD_NO_SERVICE ?
+					$this->renderOverview().($this->spacer !== null ? $this->spacer->render() : "") : "")
+				.$this->renderTable();
 	}
 
 	protected function renderOverview() {
 		$tpl = new ilTemplate("tpl.gev_edu_bio_overview.html", true, true, $this->object->plugin->getDirectory());
 		$this->insertAcademyPoints($tpl);
-		if (gevWBD::getInstance($this->object->target_user_id)->transferPointsFromWBD()) {
+		if($this->object->getWBD()->getWBDTPType() === gevWBD::WBD_TP_SERVICE ) {
 			$this->insertWBDPoints($tpl);
-			$tpl->setVariable("WBDPOINTSVISIBIBLE", "visible");
-		}
-		else {
-			$tpl->setVariable("WBDPOINTSVISIBIBLE", "invisible");
-			if(gevWBD::getInstance($this->object->target_user_id)->transferPointsToWBD()) {
-				$tpl->setVariable("WBDTRANSVISIBIBLE", "visible");
-				$tpl->setCurrentBlock("wbd_transfer");
-				$tpl->setVariable("TRANSFER_TITLE", $this->object->plugin->txt("wbd_transfer_on"));
-				$tpl->parseCurrentBlock();
-			}
-			elseif (gevWBD::getInstance($this->object->target_user_id)->wbdRegistrationIsPending()){
+			if ($this->object->getWBD()->transferPointsFromWBD() || true) {
+				$tpl->setVariable("WBDPOINTSVISIBIBLE", "visible");
+			} else {
 				$tpl->setVariable("WBDPOINTSVISIBIBLE", "invisible");
-				$tpl->setCurrentBlock("wbd_reg_pending");
-				$tpl->setVariable("WBDREGPENDINGVISIBIBLE", "visible");
-				$tpl->setVariable("WBD_REG_PENDING", $this->object->plugin->txt("wbd_reg_pending"));
-				$tpl->parseCurrentBlock();
-			}
-			else {
-				$tpl->setVariable("WBDTRANSVISIBIBLE", "visible");
+				if($this->object->getWBD()->transferPointsToWBD()) {
+					$tpl->setVariable("WBDTRANSVISIBIBLE", "visible");
+					ilUtil::sendInfo($this->plugin->txt("wbd_transfer_on"));
+				} elseif($this->object->getWBD()->wbdRegistrationIsPending()) {
+					$tpl->setVariable("WBDPOINTSVISIBIBLE", "invisible");
+					ilUtil::sendInfo($this->plugin->txt("wbd_reg_pending"));
+				} else {
+					$tpl->setVariable("WBDTRANSVISIBIBLE", "visible");
+				}
 			}
 		}
 		return $tpl->get();
 	}
 
 	protected function insertAcademyPoints($tpl) {
-		$tpl->setVariable("ACADEMY_SUM_TITLE",$this->object->plugin->txt("points_in_academy"));
-		$tpl->setVariable("ACADEMY_SUM_FIVE_YEAR_TITLE", $this->object->plugin->txt("points_in_five_years"));
-
-		$tpl->setVariable("ACADEMY_FIVE_YEAR", $this->object->academy_data["five_year"]);
-
-		if ($aux = $this->object->academy_data["sum"]) {
-			$tpl->setVariable("ACADEMY_SUM", $aux ? $aux : 0);
+		$tpl->setVariable("ACA_TRANSFERED_SUM_TITLE", $this->object->plugin->txt("aca_transferred_points_filter"));
+		$tpl->setVariable("ACA_TO_TRANSFER_SUM_TITLE", $this->object->plugin->txt("aca_to_transdfer_points_filter"));
+		if ($aux = $this->object->academy_points["transfered_sum"]) {
+			$tpl->setVariable("ACA_TRANSFERED_SUM", $aux ? $aux : 0);
 		}
```

Comment:
Könnte man hier nicht auch einfach nur schreiben:

$tpl->setVariable("ACA_TRANSFERED_SUM", $this->object->academy_points["transfered_sum"] ? $this->object->academy_points["transfered_sum"] : 0);